week02/Valid_Anagram.cpp

```diff
@@ -0,0 +1,36 @@
+#Time Complexity:O(n)
+#Space Complexity:O(1)
+
+class Solution {
+public:
+    bool isAnagram(string s, string t) {
+
+       if(s.size()!=t.size())
+           return false;
+
+        unordered_map<char,int>mp;
+
+        for(int i=0;i<s.size();i++){
+            auto it=mp.find(s[i]);
+
+            if(it==mp.end())
+                mp[s[i]]=1;
+            else
+                mp[s[i]]++;
+        }
+         for(int i=0;i<t.size();i++){
+            auto it=mp.find(t[i]);
+
+            if(it==mp.end())
+                return false;
+            else{
+                mp[t[i]]--;
+                if(mp[t[i]]<0)
+                    return false;
+            }
+
+        }
+
+        return true;
+    }
+};
```

Comment:
Lengthy and suboptimal solution. It has a more straightforward, concise, and optimal solution. Kindly refer to our [editorial](https://docs.google.com/document/d/1VipVcGyqiUDvFWbJ_yivTJucBRw3ct7HGZGGx-_CRsk/edit).